common/src/main/java/com/wynntils/features/user/overlays/MiniMapOverlayFeature.java

```diff
@@ -0,0 +1,189 @@
+import com.wynntils.core.features.overlays.annotations.OverlayInfo;
+import com.wynntils.core.features.overlays.sizes.GuiScaledOverlaySize;
+import com.wynntils.core.webapi.WebManager;
+import com.wynntils.core.webapi.profiles.MapProfile;
+import com.wynntils.mc.event.RenderEvent;
+import com.wynntils.mc.render.HorizontalAlignment;
+import com.wynntils.mc.render.RenderUtils;
+import com.wynntils.mc.render.VerticalAlignment;
+import com.wynntils.mc.utils.McUtils;
+import net.minecraft.client.renderer.GameRenderer;
+import org.lwjgl.opengl.GL11;
+import org.lwjgl.opengl.GL13;
+
+public class MiniMapOverlayFeature extends UserFeature {
+    @OverlayInfo(renderType = RenderEvent.ElementType.GUI)
+    public final MiniMapOverlay miniMapOverlay = new MiniMapOverlay();
+
+    public static class MiniMapOverlay extends Overlay {
+
+        @Config
+        public float scale = 1f;
+
+        @Config
+        public boolean followPlayerRotation = true;
+
+        @Config
+        public boolean renderUsingLinear = true;
+
+        @Config
+        public MapMaskType maskType = MapMaskType.Rectangular;
+
+        public MiniMapOverlay() {
+            super(
+                    new OverlayPosition(
+                            0,
+                            0,
+                            VerticalAlignment.Top,
+                            HorizontalAlignment.Left,
+                            OverlayPosition.AnchorSection.TopLeft),
+                    new GuiScaledOverlaySize(200, 200));
+        }
+
+        @Override
+        public void render(PoseStack poseStack, float partialTicks, Window window) {
+            if (!WebManager.isMapLoaded()) return;
+
+            MapProfile map = WebManager.getMaps().get(0);
+
+            float uScale = 1f / map.getTextureWidth();
+            float vScale = 1f / map.getTextureHeight();
+
+            float width = getWidth();
+            float height = getHeight();
+            float renderX = getRenderX();
+            float renderY = getRenderY();
+
+            float centerX = renderX + width / 2;
+            float centerZ = renderY + height / 2;
+            float textureX = map.getTextureXPosition(McUtils.player().getX());
+            float textureZ = map.getTextureZPosition(McUtils.player().getZ());
+
+            // avoid rotational overpass
+            float extraFactor = (followPlayerRotation && maskType == MapMaskType.Rectangular ? 1.5F : 1);
+
+            float halfRenderedWidth = width / 2 * extraFactor;
+            float halfRenderedHeight = height / 2 * extraFactor;
+            float halfTextureWidth = halfRenderedWidth * scale;
+            float halfTextureHeight = halfRenderedHeight * scale;
+
+            // Render Minimap
+            RenderSystem.setShader(GameRenderer::getPositionTexShader);
+            RenderSystem.setShaderTexture(0, map.resource());
+
+            // clamp map rendering
+            int option = renderUsingLinear ? GL11.GL_LINEAR : GL11.GL_NEAREST;
```

Comment:
I could not for the life of me tell the difference. Is that depending on graphics card, if it makes a difference or not? Is it worth having a user option, or should we just not just pick what we think is the best?

Reply:
I can't tell the difference either, this is a port from legacy. Apparently it does something about aliasing.

Reply:
Try zoom in or out really far with both options.

Reply:
I tried zooming in and out, but perhaps not enough.

Reply:
Now I've retried with scales ranging from 0.1 (only seeing a few pixels) to 50 (seeing the entire continent). I can see *no* difference whatsoever. But it might be my computer. I have a macbook, and the GL implementation in macOS is known to be minimalistic and sometimes not even fully standards compliant.

Can you grab some screenshots to show me how it differs for you?

Reply:
It is as far as I can tell identical, maybe the implementation is wrong?

Reply:
Given that I'm on a mac, I'm going to close this and leave this for someone else as a separate pr.